samanthas_telegram_bot/api_clients/smalltalk/smalltalk_client.py

```diff
@@ -0,0 +1,273 @@
+                            f"{total_seconds_waiting / 60} minutes. "
+                            f"Interview ID {user_data.student_smalltalk_test_id}."
+                        ),
+                        needs_to_notify_admin_group=True,
+                    )
+                    user_data.comment = (
+                        f"{user_data.comment}\n- SmallTalk assessment results were not ready\n"
+                        f"Interview ID {user_data.student_smalltalk_test_id}"
+                    )
+
+                logger.info("SmallTalk results not ready. Waiting...")
+                attempts += 1
+                await asyncio.sleep(SMALLTALK_TIMEOUT_IN_SECS_BETWEEN_API_REQUEST_ATTEMPTS)
+            else:
+                await logs(
+                    bot=context.bot,
+                    update=update,
+                    text=(
+                        f"Received [SmallTalk results for "
+                        f"{escape_for_markdown(user_data.first_name)} "
+                        f"{escape_for_markdown(user_data.last_name)}]({result.url})"
+                    ),
+                    parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+                    needs_to_notify_admin_group=True,
+                )
+                return result
+
+    @classmethod
+    async def _get_data(
+        cls, update: Update, context: CUSTOM_CONTEXT_TYPES, test_id: str
+    ) -> DataDict:
+        user_data = context.user_data
+
+        try:
+            _, data = await cls.get(
+                update=update,
+                context=context,
+                url=SMALLTALK_URL_GET_RESULTS,
+                headers=HEADERS,
+                params={
+                    "id": test_id,
+                    "additional_fields": (
+                        "detailed_scores,strength_weaknesses,problem_statuses,problem_titles"
+                    ),
+                },
+                notification_params_for_status_code={
+                    httpx.codes.OK: NotificationParams(
+                        # The data may not contain the results yet, we're just logging
+                        # that status code was 200 OK and hence some data was received
+                        f"Received data from SmallTalk after {user_data.first_name} "
+                        f"{user_data.last_name}'s test"
+                    )
+                },
+            )
+        except BaseApiClientError as err:
+            raise SmallTalkRequestError(
+                "Failed to receive valid response from SmallTalk while getting test results."
+            ) from err
+
+        return data
+
+    @classmethod
+    async def _process_data(
+        cls,
+        update: Update,
+        context: CUSTOM_CONTEXT_TYPES,
+        data: DataDict,
+    ) -> SmalltalkResult:
+        status = typing.cast(SmalltalkTestStatus, cls._get_value(data, "status"))
+
+        if status not in SmalltalkTestStatus._value2member_map_:  # noqa
```

Comment:
doesn't  just `if status not in SmalltalkTestStatus` work?

Reply:
If status is a string, you can do SmalltalkTestStatus(status) to make it enum instance. With try..except ofc

Reply:
> If status is a string, you can do SmalltalkTestStatus(status) to make it enum instance. With try..except ofc

Yep, did that. Thanks.